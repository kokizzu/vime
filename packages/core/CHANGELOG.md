# Change Log

All notable changes to this project will be documented in this file.

## [1.0.2](https://github.com/vime-js/vime/compare/@vime/core@1.0.1...@vime/core@1.0.2) (2020-08-13)


### Bug Fixes

* **core:** point unpkg and jsdelivr to correct file ([8360ad6](https://github.com/vime-js/vime/commit/8360ad62197ab35a3f51986f83e9412c7fd11a3b))

## [1.0.1](https://github.com/vime-js/vime/compare/@vime/core@1.0.0...@vime/core@1.0.1) (2020-08-13)


### Bug Fixes

* **core:** icons not loading ([2b48c85](https://github.com/vime-js/vime/commit/2b48c8547b0b7454b48a2c2707e82d74837ae9cb))

# 1.0.0 (2020-08-12)


### Bug Fixes

* **core:** lazy loading broken ([792520a](https://github.com/vime-js/vime/commit/792520ac17dcd8d70179ac4b29d91d4ac5a127eb))
* **core:** prevent posters loading twice with custom UI ([5aefb06](https://github.com/vime-js/vime/commit/5aefb06f3ad7e80204ce4966f42d5372a3517a70))
* **core/ui:** slider keydown events propagating ([94fafe1](https://github.com/vime-js/vime/commit/94fafe1f485e841febea3dd8a065588bf874622a))
* **core/ui/captions:** emitting events after disconnecting ([5f0abac](https://github.com/vime-js/vime/commit/5f0abac07f6b497607968265435e76a248ba3edf))
* **core/ui/captions:** toggling captions on/off broken ([090d776](https://github.com/vime-js/vime/commit/090d7762401750d91f9a1cc12a320cbfa3652eb6))


### Features

* **core:** add lazy loading via lozad ([3bde06c](https://github.com/vime-js/vime/commit/3bde06c5928895195f8e934c6808eca964737d07))
* **core:** add skeleton loading animation ([ff7860d](https://github.com/vime-js/vime/commit/ff7860ddfe74f029c3d365870a31d6d3d72f2748))
* **core:** dailymotion + hls + dash providers ([c520c30](https://github.com/vime-js/vime/commit/c520c302e5db91f20c9944915a43102ff1305e51))
* **core:** improve test harness + vimeo provider ([dd153ca](https://github.com/vime-js/vime/commit/dd153ca96026a3ef80c89fa1dcc560e36dcc7c1d))
* **core:** vime-embed component ([b45be6f](https://github.com/vime-js/vime/commit/b45be6f642e286b7da568b4450e8268334961019))
* **core:** vime-player component ([c86bee4](https://github.com/vime-js/vime/commit/c86bee4f709fd51d6457805df6c6c19efa0abc89))
* **core:** youtube provider + providers test harness ([8f93a68](https://github.com/vime-js/vime/commit/8f93a68edd16d4396c708e84e8e5bc1d8da69c41))
* **core/ui:** add click-to-play component ([fcce4f1](https://github.com/vime-js/vime/commit/fcce4f11b32f4603f96677dd10803233c344e8d8))
* **core/ui:** add light audio player theme ([93d4206](https://github.com/vime-js/vime/commit/93d42062ee2515ac37a10244961767714cfa87f9))
* **core/ui:** captions component ([c810f42](https://github.com/vime-js/vime/commit/c810f42d00cc81aa3cf35bf81baf2846dcbba4bb))
* **core/ui:** controls components ([5dc4e26](https://github.com/vime-js/vime/commit/5dc4e261e23df6993694abfa776af8637fb7b9a6))
* **core/ui:** create settings components ([3a1fd82](https://github.com/vime-js/vime/commit/3a1fd82a232b67f27759dc94e1e76fea8228f97e))
* **core/ui:** icons + icon component ([d13df8e](https://github.com/vime-js/vime/commit/d13df8ef1b265ec995b8a37f023162f7a978da37))
* **core/ui:** poster + spinner components ([cbdd69f](https://github.com/vime-js/vime/commit/cbdd69ff9f3474639f2a781809859c3b94beb7ff))
* **core/ui:** scrim component ([df47efe](https://github.com/vime-js/vime/commit/df47efe31b3088157896120b9f7ff2d78bdd5c99))
* **core/ui/controls:** add option to scale control size up/down ([3635b1a](https://github.com/vime-js/vime/commit/3635b1ae0d2d2d0163086a01d4e1f6fb2468a561))
* **core/ui/poster:** add poster events ([9d36f47](https://github.com/vime-js/vime/commit/9d36f47a55f6459e8ffc24018cf2fb5c2654fcde))
* **core/ui/scrim:** add events ([1180240](https://github.com/vime-js/vime/commit/1180240c8e93cce96009c3c2098064806dfa4c37))
* **core/ui/scrim:** add gradient option ([adf6bcb](https://github.com/vime-js/vime/commit/adf6bcbdb2d8520552bb6c93b5bc1f30ab95ce84))
* **core/ui/settings:** add audio/mobile styles to settings ([c937312](https://github.com/vime-js/vime/commit/c937312b344e2a5b3bb10315367f98c6b0eb81a0))
* **core/ui/spinner:** add events ([13dfd14](https://github.com/vime-js/vime/commit/13dfd1445d72357a1ed315cd3a25e955be19a772))
