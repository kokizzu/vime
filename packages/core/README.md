# @vime/core

Vime is simply a collection of [web components](https://developer.mozilla.org/en-US/docs/Web/Web_Components)
that help you easily build and customize your own media player.

The `@vime/core` package contains the web components for the player, plugins, providers and
UI elements that makeup the building blocks of Vime.

## Example

```html
<vime-player autoplay muted>
  <vime-video poster="/media/poster.png" cross-origin>
    <source data-src="/media/video.mp4" type="video/mp4" />
    <track
      default
      kind="subtitles"
      src="/media/subs/en.vtt"
      srclang="en"
      label="English"
    />
    <track
      kind="subtitles"
      src="/media/subs/es.vtt"
      srclang="es"
      label="Spanish"
    />
  </vime-video>

  <vime-ui>
    <vime-click-to-play />
    <vime-captions />
    <vime-poster />
    <vime-spinner />
    <vime-default-settings />
    <vime-controls pin="bottomLeft" active-duration="2750" full-width>
      <vime-playback-control tooltip-direction="right" />
      <vime-volume-control />
      <vime-time-progress />
      <vime-control-spacer />
      <vime-caption-control />
      <vime-pip-control keys="p" />
      <vime-settings-control />
      <vime-fullscreen-control keys="f" tooltip-direction="left" />
    </vime-controls>
  </vime-ui>

  <vime-plugins>
    <vime-chromecast />
  </vime-plugins>
</vime-player>
```

## Related

- [GitHub](https://github.com/vime-js/vime)
- [Discord](https://discord.gg/PaFFSk)
- [Documentation](https://vimejs.com)
