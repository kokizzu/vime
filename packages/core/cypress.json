{
  "baseUrl": "http://localhost:3333/src/components",
  "watchForFileChanges": false,
  "integrationFolder": "cypress/tests",
  "testFiles": "**/*.e2e.ts",
  "fixturesFolder": false,
  "supportsFolder": "cypress/support/index.ts",
  "ignoreTestFiles": [
    "**/__snapshots__/*",
    "**/__image_snapshots__/*"
  ],
  "env": {
    "cypress-plugin-snapshots": {
      "imageConfig": {
        "threshold": 0.05
      }
    }
  }
}
