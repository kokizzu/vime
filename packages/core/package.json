{
  "name": "@vime/core",
  "version": "1.11.0",
  "description": "Collection of web components to build and customize your own media player.",
  "license": "MIT",
  "main": "dist/index.cjs.js",
  "module": "dist/index.js",
  "es2015": "dist/esm/index.js",
  "es2017": "dist/esm/index.js",
  "jsnext:main": "dist/esm/index.js",
  "collection": "dist/collection/collection-manifest.json",
  "collection:main": "dist/collection/index.js",
  "types": "dist/types/index.d.ts",
  "files": [
    "dist/",
    "loader/",
    "icons/",
    "themes/"
  ],
  "contributors": [
    "Rahim Alwer <rahim_alwer@outlook.com>"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/vime-js/vime.git"
  },
  "bugs": {
    "url": "https://github.com/vime-js/vime/issues"
  },
  "scripts": {
    "build": "npm run clean:docs && stencil build --docs && npm run build:post && npm run build:themes && npm run lint:md",
    "build:post": "node scripts/post-build.js",
    "build:themes": "rimraf themes/* && npm run min:theme.default && npm run min:theme.light",
    "build:coverage": "npm run build && npm run coverage:instrument",
    "clean:docs": "rimraf ../../docs/docs/components",
    "coverage:instrument": "nyc instrument dist --in-place",
    "coverage:combine": "nyc report --reporter lcov --reporter text --report-dir coverage",
    "generate": "stencil generate",
    "generate:provider": "node scripts/new-provider.js",
    "lint:md": "npm run lint:md --prefix ../../",
    "min:theme.default": "cleancss -o themes/default.css src/globals/themes/default.css",
    "min:theme.light": "cleancss -o themes/light.css src/globals/themes/light.css",
    "serve": "concurrently \"npm run serve:public\" \"npm run serve:dev\"",
    "serve:dev": "stencil build --dev --watch --serve --port 3336",
    "serve:public": "node scripts/link-public.js && sirv public --single tests/index.html --cors --port 3334 --quiet",
    "serve:prod": "npm run build && npm run serve:public",
    "test:unit": "stencil test --spec --max-workers=2",
    "test:unit.debug": "pnpx --node-arg=\"--inspect-brk\" stencil test --spec",
    "test:unit.watch": "stencil test --spec --watchAll --max-workers=2",
    "test:e2e": "npm run build:coverage && start-server-and-test serve:public http://localhost:3334 cy:run",
    "test:e2e.ui": "npm run build && start-server-and-test serve:public http://localhost:3334 cy:run.ui",
    "test:e2e.providers": "npm run build && start-server-and-test serve:public http://localhost:3334 cy:run.providers",
    "test:ci": "rimraf .nyc_output && npm run test:unit --coverage && npm run test:e2e && npm run coverage:combine",
    "release": "semantic-release -e ../../scripts/semantic-release",
    "cy:open": "cypress open",
    "cy:run": "cypress run",
    "cy:run.ui": "cypress run --spec \"cypress/tests/ui/**/*\"",
    "cy:run.providers": "cypress run --spec \"cypress/tests/providers/**/*\""
  },
  "dependencies": {
    "stencil-wormhole": "^3.2.1"
  },
  "devDependencies": {
    "@babel/core": "^7.11.4",
    "@babel/preset-env": "^7.11.0",
    "@cypress/code-coverage": "^3.8.1",
    "@cypress/skip-test": "^2.5.0",
    "@cypress/webpack-preprocessor": "^5.4.5",
    "@stencil/angular-output-target": "^0.0.5",
    "@stencil/core": "2.0.0-8",
    "@stencil/react-output-target": "^0.0.9",
    "@stencil/sass": "^1.3.2",
    "@stencil/svelte-output-target": "^0.0.3",
    "@stencil/vue-output-target": "^0.0.3",
    "@types/hls.js": "^0.13.0",
    "@types/jest": "^26.0.10",
    "@types/puppeteer": "^3.0.1",
    "@vime/media": "^1.0.0",
    "babel-loader": "^8.1.0",
    "clean-css-cli": "^4.3.0",
    "concurrently": "^5.3.0",
    "cypress": "^4.12.1",
    "cypress-plugin-snapshots": "^1.4.4",
    "cypress-plugin-tab": "^1.0.5",
    "jest": "^26.4.2",
    "jest-watch-typeahead": "^0.6.0",
    "js-base64": "2.5.2",
    "nyc": "^15.1.0",
    "rimraf": "^3.0.2",
    "sirv-cli": "^1.0.6",
    "start-server-and-test": "^1.11.3",
    "svelte": "^3.24.1",
    "ts-loader": "^8.0.3",
    "webpack": "^4.44.1"
  },
  "publishConfig": {
    "access": "public"
  },
  "keywords": [
    "vime",
    "player",
    "media",
    "audio",
    "video",
    "html5",
    "stenciljs",
    "youtube",
    "dailymotion",
    "vimeo",
    "hls",
    "dash",
    "custom",
    "web components"
  ],
  "nyc": {
    "report-dir": "cypress-coverage"
  }
}
