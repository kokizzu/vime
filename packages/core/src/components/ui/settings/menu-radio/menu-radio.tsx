/* eslint-disable jsx-a11y/label-has-associated-control */

import {
  h, Component, Prop, Event, EventEmitter,
} from '@stencil/core';

@Component({
  tag: 'vime-menu-radio',
})
export class MenuRadio {
  /**
   * The title of the radio item displayed to the user.
   */
  @Prop() label!: string;

  /**
   * The value associated with this radio item.
   */
  @Prop() value!: string;

  /**
   * Whether the radio item is selected or not.
   */
  @Prop({ mutable: true }) checked = false;

  /**
   * This can provide additional context about the value. For example, if the option is for a set of
   * video qualities, the badge could describe whether the quality is UHD, HD etc.
   */
  @Prop() badge?: string;

  /**
   * The URL to an SVG element or fragment to load.
   */
  @Prop() checkedIcon?: string = '#vime-checkmark';

  /**
   * Emitted when the radio button is selected.
   */
  @Event() check!: EventEmitter<void>;

  private onClick() {
    this.checked = !this.checked;
    this.check.emit();
  }

  render() {
    return (
      <vime-menu-item
        label={this.label}
        checked={this.checked}
        badge={this.badge}
        checkedIcon={this.checkedIcon}
        onClick={this.onClick.bind(this)}
      />
    );
  }
}
