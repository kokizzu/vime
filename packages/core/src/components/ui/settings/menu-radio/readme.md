# vime-menu-radio

Menu radio buttons are presented in radio groups (a collection of radio buttons describing a set 
of related options). Only one radio button in a group can be selected at the same time.

## Example

```html
<vime-player>
  <!-- ... -->
  <vime-ui>
    <!-- ... -->
    <vime-settings>
      <vime-submenu label="Playback Rate">
        <vime-menu-radio-group value="1">
          <vime-menu-radio label="0.5" value ="0.5" />
          <vime-menu-radio label="Normal" value ="1" />
          <vime-menu-radio label="2" value ="2" />
        </vime-menu-radio-group>
      </vime-submenu>
    </vime-settings>
  </vime-ui>
</vime-player>
```

<!-- Auto Generated Below -->


## Properties

| Property             | Attribute      | Description                                                                                                                                                                   | Type                  | Default             |
| -------------------- | -------------- | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | --------------------- | ------------------- |
| `badge`              | `badge`        | This can provide additional context about the value. For example, if the option is for a set of video qualities, the badge could describe whether the quality is UHD, HD etc. | `string \| undefined` | `undefined`         |
| `checked`            | `checked`      | Whether the radio item is selected or not.                                                                                                                                    | `boolean`             | `false`             |
| `checkedIcon`        | `checked-icon` | The URL to an SVG element or fragment to load.                                                                                                                                | `string \| undefined` | `'#vime-checkmark'` |
| `label` _(required)_ | `label`        | The title of the radio item displayed to the user.                                                                                                                            | `string`              | `undefined`         |
| `value` _(required)_ | `value`        | The value associated with this radio item.                                                                                                                                    | `string`              | `undefined`         |


## Events

| Event    | Description                                | Type                |
| -------- | ------------------------------------------ | ------------------- |
| `vCheck` | Emitted when the radio button is selected. | `CustomEvent<void>` |


## Dependencies

### Used by

 - [vime-default-settings](../default-settings)

### Depends on

- [vime-menu-item](../menu-item)

### Graph
```mermaid
graph TD;
  vime-menu-radio --> vime-menu-item
  vime-menu-item --> vime-icon
  vime-default-settings --> vime-menu-radio
  style vime-menu-radio fill:#f9f,stroke:#333,stroke-width:4px
```

----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
