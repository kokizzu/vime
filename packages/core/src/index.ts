import { PlayerProp, PlayerProps } from './components/core/player/PlayerProp';
import { PlayerDispatcher, createPlayerDispatcher } from './components/core/player/PlayerDispatcher';
import { withPlayerContext, usePlayerContext } from './components/core/player/PlayerContext';
import { PlayerEvent } from './components/core/player/PlayerEvent';
import { findRootPlayer } from './components/core/player/utils';

export * from './components';

export {
  PlayerProp,
  PlayerProps,
  PlayerEvent,
  findRootPlayer,
  withPlayerContext,
  usePlayerContext,
  PlayerDispatcher,
  createPlayerDispatcher,
};
