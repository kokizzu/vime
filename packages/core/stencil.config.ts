import { Config } from '@stencil/core';
import { sass } from '@stencil/sass';

export const config: Config = {
  namespace: 'Vime',
  taskQueue: 'async',
  buildEs5: false,
  globalStyle: 'src/globals/theme.css',
  extras: {
    cssVarsShim: false,
    dynamicImportShim: false,
    safari10: false,
    scriptDataOpts: false,
    shadowDomShim: false,
  },
  plugins: [
    sass({
      injectGlobalPaths: [
        'src/globals/variables.scss',
        'src/globals/mixins.scss',
      ],
    }),
  ],
  outputTargets: [
    {
      type: 'dist',
      esmLoaderPath: '../loader',
    },
    {
      type: 'docs-readme',
    },
  ],
  testing: {
    globals: {
      'ts-jest': {
        diagnostics: false,
      },
    },
    setupFilesAfterEnv: [
      './src/globals/jest.js',
    ],
    coverageDirectory: 'jest-coverage',
    // @ts-ignore
    watchPlugins: [
      'jest-watch-typeahead/filename',
      'jest-watch-typeahead/testname',
    ],
    testPathIgnorePatterns: [
      '/node_modules/',
      '/cypress/',
    ],
  },
  preamble: '(C) Vime https://vime-js.com - MIT License',
};
