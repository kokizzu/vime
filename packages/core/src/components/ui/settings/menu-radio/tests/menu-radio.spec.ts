import { SpecPage, newSpecPage } from '@stencil/core/testing';
import { MenuRadio } from '../menu-radio';

let page: SpecPage;

beforeEach(async () => {
  page = await newSpecPage({
    components: [MenuRadio],
    html: '<vime-menu-radio />',
  });
});

it('should be structurally sound', () => {
  expect(page.root).toMatchSnapshot();
});
