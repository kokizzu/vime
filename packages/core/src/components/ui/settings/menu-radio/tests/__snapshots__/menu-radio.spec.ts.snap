// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be structurally sound 1`] = `
<vime-menu-radio>
  <vime-menu-item checkedicon="#vime-checkmark"></vime-menu-item>
</vime-menu-radio>
`;
